dom.bs

```diff
@@ -2645,12 +2645,72 @@ of a <var>node</var> into a <var>parent</var> before a <var>child</var>, run the
+do these things asynchronously, however.
+
+<div class=example id=example-foo-what-do-i-put-here>
+ <p>While the <a>insertion steps</a> cannot execute JavaScript (among other things), it is expected
+ that they will indeed have script-observable consequences. Consider the below example:
+
+ <pre class=lang-javascript><code>
+ const h1 = document.querySelector('h1');
+
+ const fragment = new DocumentFragment();
+ const script = fragment.appendChild(document.createElement('script'));
+ const style = fragment.appendChild(document.createElement('style'));
+
+ script.innerText= 'console.log(getComputedStyle(h1).color)'; // Prints 'rgb(255, 0, 0)'
+ style.innerText = 'h1 {color: rgb(255, 0, 0);}';
+
+ document.body.append(fragment);
+ </code></pre>
+
+ <p>The script in the above example prints <code class=lang-javascript>'rgb(255, 0, 0)'</code>
+ because the following happen in order:
+
+  <ol>
+   <li>
+    <p>The <a for=/>insert</a> algorithm runs, which will insert the <{script}> and <code><a element
+    spec=HTML>style</a></code> elements in order.
+
+    <ol>
+     <li>The HTML Standard's <a>insertion steps</a> run for the <{script}> element; they do nothing.
+     [[!HTML]]
+
+     <li>The HTML Standard's <a>insertion steps</a> run for the <code><a element
+     spec=HTML>style</a></code> element; they immediately apply its style rules to the document.
+     [[!HTML]]
+
+     <li>The HTML Standard's <a>post-insertion steps</a> run for the <{script}> element; they run
+     the script, which immediately observes the style rules that were applied in the above step.
+     [[!HTML]]
+    </ol>
+   </li>
+  </ol>
+</div>
+
 <!-- See https://github.com/whatwg/dom/issues/34#issuecomment-125571750 for why we might need to
      adjust this further based on the requirements of the script element. There might be other ways
      to define that though as Olli suggests, so leaving that out for now. -->
 
+<p><a lt="Other applicable specifications">Specifications</a> may also define <dfn export
+id=concept-node-post-insert-ext>post-insertion steps</dfn> for all or some <a for=/>nodes</a>. The
+algorithm is passed <var ignore>insertedNode</var>, as indicated in the <a
+for=/>insert</a> algorithm below.
+
+<p class=note>The purpose of the <a>post-insertion steps</a> is to provide an opportunity for <a
+for=/>nodes</a> to perform any insertion-related operations that modify the <a>node tree</a> that
+<var ignore>insertedNode</var> <a>participates</a> in, create <a for=/>browsing contexts</a>, or
+otherwise execute JavaScript. These steps allow a batch of <a>nodes</a> to be <a for=/>inserted</a>
+<i>atomically</i> with respect to script, with all major side effects occurring <i>after</i> the
+batch insertions into the <a>node tree</a> is complete, but before <a lt="notify mutation
+observers"><code>MutationObserver</code>s are notified</a>. This ensures that all pending <a>node
```

Comment:
Why is this talking about notifying MutationObservers? That happens at microtask checkpoint which is way later.

Reply:
I added this in https://github.com/whatwg/dom/pull/1261/commits/164eae9b4bc2ea64a8a7426d6ac82e55c9904639, in response to https://github.com/whatwg/dom/pull/1261#discussion_r1520204437 which mentioned MutationObservers. Idk, for this non-normative text it seemed reasonable to just drive home the point that this all happens technically after insertion, but still very much synchronously, and therefore earlier than the next-soonest reaction timing thing, which is MutationObserver firing. If you feel strongly, I'm happy to remove if Anne agrees.

Reply:
CE Reaction or what? That isn't when MutationObservers get notified.

Reply:
Ah, well we also have tests for [custom element reactions](https://wpt.fyi/results/dom/nodes/insertion-removing-steps/Node-appendChild-script-and-custom-from-fragment.tentative.html?label=experimental&label=master&aligned). In any case, my point is that the non-normative text here is just to drive home the point that the post-insertion steps run synchronously after insertion and before anything. Do you think I should remove it?

Reply:
I think it is very confusing to talk about MutationObservers (microtasks) here.

Reply:
@annevk what do you think? I think mentioning the fact that post-insertion steps run synchronously before any microtasks run is useful, but if you agree with Olli I'm happy to nix this bit.